components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -163,9 +157,22 @@ func main() {
 		broker.NewLastBindingOperation(dumper),
 	}
 
-	// create and run broker OSB API
-	brokerAPI := brokerapi.New(kymaEnvBroker, logger, brokerCredentials)
-	r := handlers.LoggingHandler(os.Stdout, brokerAPI)
+	// create broker credentials
+	brokerCredentials := broker.BrokerCredentials{
+		Username: cfg.Auth.Username,
+		Password: cfg.Auth.Password,
+	}
+
+	// create and run broker OSB API in 2 modes:
+	// with basic auth
+	// with oauth
+	brokerBasicAPI := broker.New("", kymaEnvBroker, logger, &brokerCredentials)
+	brokerAPI := broker.New("/oauth", kymaEnvBroker, logger, nil)
+	n := negroni.New(negroni.NewRecovery())
```

Comment:
with that solution I'm getting 404 when calling the https://kyma-env-broker.$domain/v2/catalog

and such response body:
```
404 page not found
Not Authorized
```

**expected:**
code: 401
body: "Not Authorized"

When I provide a problem basic then I got such reponse:
```
404 page not found
{"services":[{"id":"47c9dcbf-ff30-448e-ab36-d3bad66ba281","name":"kymaruntime","description":"[EXPERIMENTAL] Service Class for Kyma Runtime","bindable":true,"tags":["SAP","Kyma"],"plan_updateable":false,"plans":[{"id":"ca6e5357-707f-4565-bbbd-b3ab732597c6","name":"gcp","description":"GCP","metadata":{"displayName":"GCP"},"schemas":{"service_instance":{"create":{"parameters":{"$schema":"http://json-schema.org/draft-04/schema#","properties":{"autoScalerMax":{"type":"integer"},"autoScalerMin":{"type":"integer"},"components":{"items":{"enum":["Kiali","Jaeger"],"type":"string"},"type":"array"},"diskType":{"type":"string"},"machineType":{"enum":["n1-standard-2","n1-standard-4","n1-standard-8","n1-standard-16","n1-standard-32","n1-standard-64"],"type":"string"},"maxSurge":{"type":"integer"},"maxUnavailable":{"type":"integer"},"name":{"type":"string"},"region":{"enum":["asia-south1","asia-southeast1","asia-east2","asia-east1","asia-northeast1","asia-northeast2","australia-southeast1","europe-west2","europe-west1","europe-west4","europe-west6","europe-west3","europe-north1","us-west1","us-west2","us-central1","us-east1","us-east4","northamerica-northeast1","southamerica-east1"],"type":"string"},"volumeSizeGb":{"type":"integer"},"zone":{"enum":["asia-east1-a","asia-east1-b","asia-east1-c","asia-east2-a","asia-east2-b","asia-east2-c","asia-northeast1-a","asia-northeast1-b","asia-northeast1-c","asia-northeast2-a","asia-northeast2-b","asia-northeast2-c","asia-south1-a","asia-south1-b","asia-south1-c","asia-southeast1-a","asia-southeast1-b","asia-southeast1-c","australia-southeast1-a","australia-southeast1-b","australia-southeast1-c","europe-north1-a","europe-north1-c","europe-north1-b","europe-west1-b","europe-west1-c","europe-west1-d","europe-west2-a","europe-west2-b","europe-west2-c","europe-west3-a","europe-west3-b","europe-west3-c","europe-west4-a","europe-west4-b","europe-west4-c","europe-west6-a","europe-west6-b","europe-west6-c","northamerica-northeast1-a","northamerica-northeast1-b","northamerica-northeast1-c","southamerica-east1-a","southamerica-east1-b","southamerica-east1-c","us-central1-a","us-central1-b","us-central1-c","us-central1-f","us-east1-b","us-east1-c","us-east1-d","us-east4-a","us-east4-b","us-east4-c","us-west1-a","us-west1-b","us-west1-c","us-west2-a","us-west2-b","us-west2-c"],"type":"string"}},"required":["name"],"type":"object"}},"update":{"parameters":null}},"service_binding":{"create":{"parameters":null}}}},{"id":"4deee563-e5ec-4731-b9b1-53b42d855f0c","name":"azure","description":"Azure","metadata":{"displayName":"Azure"},"schemas":{"service_instance":{"create":{"parameters":{"$schema":"http://json-schema.org/draft-04/schema#","properties":{"autoScalerMax":{"type":"integer"},"autoScalerMin":{"type":"integer"},"components":{"items":{"enum":["Jaeger","Kiali"],"type":"string"},"type":"array"},"diskType":{"type":"string"},"machineType":{"enum":["Standard_D8_v3"],"type":"string"},"maxSurge":{"type":"integer"},"maxUnavailable":{"type":"integer"},"name":{"type":"string"},"region":{"enum":["westeurope","eastus","eastus2","centralus","northeurope","southeastasia","japaneast","westus2","uksouth"],"type":"string"},"volumeSizeGb":{"minimum":50,"type":"integer"},"zone":{"type":"string"}},"required":["name"],"type":"object"}},"update":{"parameters":null}},"service_binding":{"create":{"parameters":null}}}}],"metadata":{"displayName":"Kyma Runtime","documentationUrl":"kyma-project.io","longDescription":"Kyma Runtime experimental service class","providerDisplayName":"SAP"}}]}
```

with code 404 and content type `text/plain; charset=utf-8`


---

oauth endpoint works properly.

Reply:
Fixed